model/publishing/delivery/RemotePublishingService.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+declare(strict_types=1);
+
+namespace oat\taoPublishing\model\publishing\delivery;
+
+use Throwable;
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\tao\model\taskQueue\Task\CallbackTaskInterface;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoPublishing\model\publishing\delivery\tasks\DeployTestEnvironments;
+use oat\taoPublishing\model\publishing\exception\PublishingFailedException;
+
+class RemotePublishingService extends ConfigurableService
+{
+    use OntologyAwareTrait;
+
+    /** @var string */
+    private $deliveryUri;
+
+    /** @var core_kernel_classes_Resource */
+    private $deliveryResource;
+
+    /** @var string */
+    private $testUri;
+
+    /** @var QueueDispatcherInterface */
+    private $queueDispatcher;
+
+    /**
+     * @param string $deliveryUri
+     * @param array $environments
+     * @return CallbackTaskInterface[]
+     * @throws PublishingFailedException
+     */
+    public function publishDeliveryToEnvironments(string $deliveryUri, array $environments): array
```

Comment:
Environment might be disabled, it should be checked before creating the task and during execution of the task as well.

Reply:
The same, it will be implemented when related branch will be merged. For the moment we do not have what to check.